src/pam_ibmacf.cpp

```diff
@@ -266,38 +261,49 @@ void setSerialNumberProperty(const string& obj){
     mSerialNumber = obj;
 }
 #else
-int readFieldModeProperty(const string& obj, const string& inf,
-                          const string& prop, pam_handle_t* pamh)
+int readFieldMode(pam_handle_t* pamh)
 {
-    bool propBool = false;
-    std::string object = obj;
-    auto bus = sdbusplus::bus::new_default();
-    auto properties = bus.new_method_call(
-        "xyz.openbmc_project.Software.BMC.Updater", object.c_str(),
-        "org.freedesktop.DBus.Properties", "Get");
-    properties.append(inf);
-    properties.append(prop);
-    try
+    FILE* pipe = popen("fw_printenv -n fieldmode 2>&1", "r");
+    if (!pipe)
     {
-        auto result = bus.call(properties);
-        if (!result.is_method_error())
+        pam_syslog(pamh, LOG_ERR, "popen failed\n");
+        return PAM_SYSTEM_ERR;
+    }
+    std::array<char, 512> buffer;
+    std::stringstream result;
+    while (fgets(buffer.data(), buffer.size(), pipe) != nullptr)
+    {
+        result << buffer.data();
+    }
+    int stat = pclose(pipe);
+    if (WIFEXITED(stat))
+    {
+        // Handle expected results
+        int rc = WEXITSTATUS(stat);
+        if (rc == 0)
         {
-            std::variant<bool> val{false};
-            result.read(val);
-            if (auto pVal = std::get_if<bool>(&val))
+            // fw_printenv exited normally with rc=0
+            if (0 == strcmp(result.str().c_str(), "true\n"))
             {
-                pam_syslog(pamh, LOG_DEBUG, "FieldModeProperty = %d\n", (*pVal));
-                propBool = (*pVal);
+                pam_syslog(pamh, LOG_ERR, "fieldmode=true");
```

Comment:
Why the LOG_ERR here for good path?

Reply:
Good catch!  I changed it for debug, and forgot to change it back.